src/test/scala/com/databricks/spark/xml/XmlSuite.scala

```diff
@@ -474,6 +475,26 @@ final class XmlSuite extends AnyFunSuite with BeforeAndAfterAll {
     assert(booksCopy.collect.map(_.toString).toSet === books.collect.map(_.toString).toSet)
   }
 
+  test("DSL save with header defined") {
```

Comment:
I apologize, I think you'll have to rebase this PR. I merged a change just now that touches the same code.
But as an upshot, there's a test case I just added which you could copy to test the first line of the output - I think that's really what you want to test, not just that it reads back. Because that would work even if this didn't write a header.
See `  test("rootTag with simple attributes") {`